agent/agentpb/subscribe.proto

```diff
@@ -0,0 +1,185 @@
+/**
+ * Subscribe represents our "streaming" subscription protocol.
+ *
+ * There is only one important RPC currently which models a generic description
+ * to a topic-based pub-sub system. The state store published updates as Events.
+ * The majority of this file is defining those event types.
+ */
+syntax = "proto3";
+
+package agentpb;
+
+import "structs.proto";
+// Go Modules now includes the version in the filepath for packages within GOPATH/pkg/mode
+// Therefore unless we want to hardcode a version here like
+// github.com/gogo/protobuf@v1.3.0/gogoproto/gogo.proto then the only other choice is to
+// have a more relative import and pass the right import path to protoc. I don't like it
+// but its necessary.
+import "gogoproto/gogo.proto";
+
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
+
+// Topic enumerates the supported streaming topics in Consul.
+enum Topic {
+    ServiceHealth = 0;
```

Comment:
When I used to do more protobuf RPC authoring, one thing that was considered a good idea was to avoid having the zero value for an enum be valid. So for example you could have the zero value for Topic be something like `Unknown = 0;` so that an omission of the topic due to bug or whatever doesn't silently operate as a `ServiceHealth` topic.